### ISSUE TITLE
Circular reference prevents install via pipx

### ISSUE THREAD

**Issue by landonb** [rec/editor#7]
Hi, I tried using `editor` in an app, but `pipx install` fails because of a circular reference between `xmod` and `dek`.

- The [`xmod` config](https://github.com/rec/xmod/blob/main/pyproject.toml) depends on `dek`:

  ```
  [tool.poetry.dependencies]
  dek = "*"
  ```

- But the [`dek` config](https://github.com/rec/dek/blob/main/pyproject.toml) depends on `xmod`:

  ```
  [tool.poetry.dependencies]
  xmod = "*"
  ```

Consequently, `pipx install <my-app>` fails, showing the error:

  ```
  $ pipx install $my_app
  ⣟ installing app ... Fatal error from pip prevented installation. Full pip output in file:
    /home/user/.local/state/pipx/log/cmd_2023-12-27_16.17.02_pip_errors.log

  pip seemed to fail to build package:
      dek

  Some possibly relevant errors from pip install:
      RecursionError: maximum recursion depth exceeded
  ```

Thanks for your hard work on your libraries! I was happy to find `editor` as a drop-in replacement for the old [`python-editor`](https://github.com/fmoo/python-editor) that doesn't work with the latest Pythons.

**Comment by rec** [comment]
Thanks for catching this, I'll fix this this morning!

**Comment by rec** [comment]
Fixed in [v1.4.1](https://github.com/rec/editor/commit/771e8c5386f1abbfeb887d51620b96d772262b1f), give it a go!

I moved everything to using poetry earlier this year and I added a dependency that wasn't necessary and worse, was circular.  

I'm sort of baffled how everything worked, to be honest, and it's a little disturbing that poetry didn't catch it.  When I get a chance I'm going to add `pipx` into my testing.

Thanks for the very clear bug report which did most of my work for me and have an excellent holiday season!
